runner/operator/tempo_nextflow_operator/tempo_nextflow_operator.py

```diff
@@ -0,0 +1,410 @@
+
+            name = "Tempo Run T:{tumor}, N:{normal}: {run_date}".format(
+                tumor=pairing["tumor"], normal=pairing["normal"], run_date=run_date
+            )
+
+            input_json = {"pairing": pairing, "mapping": mapping}
+
+            job_json = {"name": name, "app": app, "inputs": input_json, "tags": tags}
+            jobs.append(job_json)
+
+        # self.send_message(
+        #     """
+        #     Writing files to {file_path}.
+        #
+        #     Run Date: {run_date}
+        #     Beagle Version: {beagle_version}
+        #     """.format(
+        #         file_path=self.OUTPUT_DIR, run_date=run_date, beagle_version=beagle_version
+        #     )
+        # )
+
+        return [RunCreator(**job) for job in jobs]
+
+    def get_pairing_for_sample(self, tumor, pairing):
+        for p in pairing:
+            if p["tumor"] == tumor:
+                return p
+
+    def get_mapping_for_sample(self, tumor, normal, mapping):
+        map = []
+        for m in mapping:
+            if m["sample"] == tumor:
+                map.append(m)
+            if m["sample"] == normal:
+                map.append(m)
+        return map
+
+    def load_pairing_file(self, tsv_file):
+        pairing = dict()
+        with open(tsv_file, "r") as pairing_file:
+            reader = csv.reader(pairing_file, delimiter="\t")
+            next(reader, None)  # consume header
+            for row in reader:
+                try:
+                    tumor_id = row[1]
+                    normal_id = row[0]
+                    if tumor_id not in pairing:
+                        pairing[tumor_id] = normal_id
+                    else:
+                        LOGGER.error("Duplicate pairing found for %s", tumor_id)
+                except:
+                    LOGGER.error("Pairing could not be found from file for row.")
+        return pairing
+
+    def write_to_file(self, fname, s):
+        """
+        Writes file to temporary location, then registers it to the temp file group
+        Also uploads it to notifier if there is a job group id
+        """
+        output = os.path.join(self.OUTPUT_DIR, fname)
+        with open(output, "w+") as fh:
+            fh.write(s)
+        os.chmod(output, 0o777)
+        # self.register_tmp_file(output)
+        # if self.job_group_notifier_id:
+        #     upload_file_event = UploadAttachmentEvent(self.job_group_notifier_id, fname, s).to_dict()
+        #     send_notification.delay(upload_file_event)
+        # return {"class": "File", "location": "juno://" + output}
+
+    def write_historical_pairing_file(self, pairing_file_str):
+        """
+        Writes file to temporary location, then registers it to the temp file group
+        Also uploads it to notifier if there is a job group id
+        """
+        output = PAIRING_FILE_LOCATION
+        with open(output, "w+") as fh:
+            fh.write(pairing_file_str)
+        os.chmod(output, 0o777)
```

Comment:
The output of this function may need to be separated out into its own object/database entries because Tempo team may rely on it.